src/wallet/rpcwallet.cpp

```diff
@@ -111,13 +111,32 @@ bool EnsureWalletIsAvailable(const CWallet* pwallet, bool avoidException)
         "Wallet file not specified (must request wallet RPC through /wallet/<filename> uri-path).");
 }
 
+bool EnsureWalletRefIsAvailable(const std::shared_ptr<CWallet> pwallet, bool avoidException)
```

Comment:
Instead of new method, you could change `EnsureWalletIsAvailable` and update rpcdump.cpp? Also, `const std::shared_ptr<CWallet>&` avoids incr/decr usage count.

Reply:
Yes, I could, but I didn't want to affect anything outside `rpcwallet.cpp`. At least not without consultation with others.

Reply:
You can consider rpcdump.cpp and rpcwallet.cpp as one, these belong to libbitcoin_wallet.

Reply:
Ok, then I will adapt `rpcdump.cpp` as well.